server/jobs/scores_notify.py

```diff
@@ -0,0 +1,72 @@
+import os
+from collections import defaultdict
+
+import server
+
+from server import jobs
+from server.constants import STUDENT_ROLE
+from server.models import Assignment, Enrollment, User, Score, db
+from server.utils import send_email
+
+@jobs.background_job
+def email_scores(assignment_id, score_tags, subject, body,
+                 reply_to=None, dry_run=False):
+    log = jobs.get_job_logger()
+    job_creator = jobs.get_current_job().user
+
+    assign = Assignment.query.get(assignment_id)
+
+    students = [e.user for e in (Enrollment.query
+                        .options(db.joinedload('user'))
+                        .filter(Enrollment.role == STUDENT_ROLE,
+                                Enrollment.course == assign.course)
+                        .all())]
+
+    for kind in score_tags:
+        if kind not in assign.published_scores:
+            log.warning(("{0} scores are not visible to students. "
+                         " Please publish them and try again").format(kind))
+            return "Not sent - {} is not published".format(kind)
```

Comment:
This is to avoid accidentally sending a score that's not ready to be seen yet. 

We could add an override if that use case comes up.